deeprank2/molstruct/residue.py

```diff
@@ -32,58 +35,51 @@ def __init__(
         self._atoms = []
 
     def __eq__(self, other) -> bool:
-        return (
-            isinstance(self, type(other))
-            and self._chain == other._chain
-            and self._number == other._number
-            and self._insertion_code == other._insertion_code
-        )
+        if isinstance(other, Residue):
+            return (self._chain == other._chain
+                    and self._number == other._number
+                    and self._insertion_code == other._insertion_code
+            )
+        return NotImplemented
 
     def __hash__(self) -> hash:
         return hash((self._number, self._insertion_code))
 
     def get_pssm(self) -> PssmRow:
-        """
-        If the residue's chain has pssm info linked to it,
-        then return the part that belongs to this residue.
-        """
-
+        """Load pssm info linked to the residue."""
         pssm = self._chain.pssm
         if pssm is None:
             raise FileNotFoundError(f'No pssm file found for Chain {self._chain}.')
-
         return pssm[self]
 
     @property
     def number(self) -> int:
         return self._number
 
     @property
-    def chain(self):
+    def chain(self) -> Chain:
         return self._chain
 
     @property
     def amino_acid(self) -> AminoAcid:
         return self._amino_acid
 
     @property
-    def atoms(self):
+    def atoms(self) -> list["Atom"]:
```

Comment:
Why is Atom within quotation marks?

Reply:
yes. In python 3.11 it wouldn't be necessary. We could do `from __future__ import annotations` to avoid the quotation marks ([same link as above](https://adamj.eu/tech/2021/05/13/python-type-hints-how-to-fix-circular-imports/)) , but I'm not sure whether that has any other effects, so kept it like this.

Maybe importing annotations is nicer though, what do you think?